x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -198,14 +223,62 @@ export function XYChart({ data, args, formatFactory, timeZone, chartTheme }: XYC
         showLegendExtra={false}
         theme={chartTheme}
         rotation={shouldRotate ? 90 : 0}
-        xDomain={
-          data.dateRange && layers.every(l => l.xScaleType === 'time')
-            ? {
-                min: data.dateRange.fromDate.getTime(),
-                max: data.dateRange.toDate.getTime(),
-              }
-            : undefined
-        }
+        xDomain={xDomain}
+        onElementClick={([[geometry, series]]) => {
+          const xySeries = series as XYChartSeriesIdentifier;
+          const xyGeometry = geometry as GeometryValue;
+
+          const layer = layers.find(l =>
+            xySeries.seriesKeys.some((key: string | number) => l.accessors.includes(key.toString()))
+          );
+          if (!layer) {
+            return;
+          }
+
+          const table = data.tables[layer.layerId];
+
+          const points = [
+            {
+              row: table.rows.findIndex(
+                row => layer.xAccessor && row[layer.xAccessor] === xyGeometry.x
+              ),
+              column: table.columns.findIndex(col => col.id === layer.xAccessor),
+              value: (xyGeometry as GeometryValue).x,
+            },
+          ];
+
+          if (xySeries.seriesKeys.length > 1) {
+            const pointValue = xySeries.seriesKeys[0];
+
+            points.push({
+              row: table.rows.findIndex(
+                row => layer.splitAccessor && row[layer.splitAccessor] === pointValue
+              ),
+              column: table.columns.findIndex(col => col.id === layer.splitAccessor),
+              value: pointValue,
+            });
+          }
+
+          const xAxisFieldName = table.columns.find(col => col.id === layer.xAccessor)?.meta
```

Comment:
Best example of why `unknown` should always be preferred over `any`. Since `meta` is `any` we suddenly start spreading `any` here, and everything accessing `xAxisFieldName` will suddenly become `any` too.

Let's type here explicitly to prevent that: `const xAxisFieldName: string | undefined = ...`.